arangod/Aql/ModificationExecutorAccumulator.h

```diff
@@ -58,12 +60,10 @@ class ModificationExecutorAccumulator {
     _accumulator.openArray();
   }
 
-  size_t nrOfDocuments() const { return _accumulator.slice().length(); }
+  [[nodiscard]] size_t nrOfDocuments() const { return _accumulator.slice().length(); }
```

Comment:
I think this is correct.
But shall we add an assertion here that _accumulator is a closed array whenever this method is called?
For instance this is broken as long as we are still accumulating things, as we will not be able to get the Slice, which will throw a "builder not yet sealed" which is hard to find.

Reply:
added an assertion